pkg/collect/host_run.go

```diff
@@ -66,14 +85,94 @@ func (c *CollectHostRun) Collect(progressChan chan<- interface{}) (map[string][]
 		return nil, errors.Wrap(err, "failed to marshal run host result")
 	}
 
-	output := NewResult()
 	output.SaveResult(c.BundlePath, resultInfo, bytes.NewBuffer(b))
 	output.SaveResult(c.BundlePath, result, bytes.NewBuffer(stdout.Bytes()))
+	// walkthrough the output directory and save result for each file
+	if runHostCollector.OutputDir != "" {
+		runInfo.OutputDir = runHostCollector.OutputDir
+		bundleOutputRelativePath = filepath.Join(collectorRelativePath, runHostCollector.OutputDir)
+		output.SaveResults(c.BundlePath, bundleOutputRelativePath, cmdOutputTempDir)
+	}
+
+	return output, nil
+}
+
+func (c *CollectHostRun) processEnvVars(cmd *exec.Cmd) (string, error) {
+	var (
+		cmdOutputTempDir string
+		err              error
+	)
+	// clears the parent env vars
+	cmd.Env = []string{}
+	guaranteedEnvs := []string{"PATH", "KUBECONFIG"}
+	for _, key := range guaranteedEnvs {
+		guaranteedEnvVal, found := os.LookupEnv(key)
+		if found {
+			cmd.Env = append(cmd.Env,
+				fmt.Sprintf("%s=%s", key, guaranteedEnvVal))
+		}
+	}
+
+	runHostCollector := c.hostCollector
+	if runHostCollector.InheritParentEnvs {
+		if runHostCollector.InheritEnvs != nil {
+			for _, key := range runHostCollector.InheritEnvs {
+				envVal, found := os.LookupEnv(key)
+				if !found {
+					return "", errors.New(fmt.Sprintf("inherit env variable is not found: %s", key))
+				}
+				cmd.Env = append(cmd.Env,
+					fmt.Sprintf("%s=%s", key, envVal))
+			}
+		} else {
+			cmd.Env = append(cmd.Env, os.Environ()...)
+		}
+	}
+
+	if runHostCollector.Env != nil {
+		for i := range runHostCollector.Env {
```

Comment:
My custom envs have an extra `=` sign at the end

```yaml
        env:
          - AWS_REGION=us-east-1
          - AWS_ACCESS_KEY_ID={{ aws.key-id.template }}
          - AWS_SECRET_ACCESS_KEY={{ aws.secret.template }}
```

Output of `engv | grep AWS` in my custom script
```
AWS_REGION=us-east-1=
AWS_SECRET_ACCESS_KEY={{ aws.secret.template }}=
AWS_ACCESS_KEY_ID={{ aws.key-id.template }}=
```

Reply:
good catch!!

Reply:
fixed

Reply:
fixed [here](https://github.com/replicatedhq/troubleshoot/pull/1288/commits/e561a2b86387a84c100034f064b7239223e85cee#diff-533bcc91b87f8c262408a8d74f39eecbf2bd51fd01bc56ba38bfce0582011caaR166)